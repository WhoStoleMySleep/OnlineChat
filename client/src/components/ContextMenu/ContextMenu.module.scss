@import '../../styles/variables';

.context-menu {
  background: $blackBackground;
  border-radius: 4px 10px 10px 10px;
  padding: 10px 0;
  position: absolute;
  visibility: hidden;
  z-index: 99999999;

  &.active {
    visibility: visible;
  }

  &.reversed {
    border-radius: 10px 4px 10px 10px;
  }

  &__link {
    color: $buttonTextColor;
    display: block;
    font-weight: bold;
    height: 48px;
    line-height: 47px;
    padding: 0 40px;
    text-decoration: none;

    &:hover {
      background-color: rgba(255,255,255, 0.1);
    }
  }
}

.fa-pencil,
.fa-times {
  
  speak: never;

  /* Better Font Rendering =========== */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  /* use !important to prevent issues with browser extensions that change fonts */
  font-family: 'font-awesome' !important;
  font-size: 14px;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  margin-right: 8px;
  text-transform: none;
}

.fa-times:before {
  content: "\f00d";
}
.fa-pencil:before {
  content: "\f040";
}
