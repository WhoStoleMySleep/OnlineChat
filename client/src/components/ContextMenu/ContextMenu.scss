.context-menu {
  visibility: hidden;
  position: absolute;
  z-index: 99999999;
  background: rgba(0, 0, 0, 0.6);
  border-radius: 4px 10px 10px 10px;

  &.active {
    visibility: visible;
  }

  &.reversed {
    border-radius: 10px 4px 10px 10px;
  }

  &__item-list {
    list-style-type: none;
    padding: 0;
    text-align: left;
  }

  &__link {
    display: block;
    padding: 0 40px;
    color: rgb(255, 249, 213);
    text-decoration: none;
    font-weight: bold;
    height: 48px;
    line-height: 47px;

    &:hover {
      background-color: rgba(255,255,255, 0.1);
    }
  }
}

.fa-pencil,
.fa-times {
  font-size: 14px;
  margin-right: 8px;
}