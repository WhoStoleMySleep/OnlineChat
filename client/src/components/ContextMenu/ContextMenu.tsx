import React from 'react';
import './ContextMenu.scss';

function ContextMenu() {
  return (
    <nav className="context-menu">
      <ul className="context-menu__item-list">
        <li className="context-menu__item">
          <a href="##" className="context-menu__link">
            <i className="fa-pencil" />
            Edit
          </a>
        </li>
        {/* <li className="context-menu__item">
          <a href="##" className="context-menu__link">
            <i className="fa-times" />
            Delete
          </a>
        </li> */}
      </ul>
    </nav>
  );
}

export default ContextMenu;
