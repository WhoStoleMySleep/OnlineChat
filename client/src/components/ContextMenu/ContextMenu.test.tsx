import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import React from 'react';
import ContextMenu from './ContextMenu';

describe('ContextMenu', () => {
  it('Snapshot test', () => {
    render(
      <ContextMenu />
    );

    const contextMenu = screen.getByTestId('context-menu');

    expect(contextMenu).toMatchSnapshot();
  });
});
