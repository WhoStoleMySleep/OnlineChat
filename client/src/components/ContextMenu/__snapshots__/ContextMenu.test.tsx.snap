// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenu Snapshot test 1`] = `
<nav
  class="context-menu"
  data-testid="context-menu"
>
  <ul
    class="context-menu__item-list"
  >
    <li
      class="context-menu__item"
    >
      <a
        class="context-menu__link"
        href="##"
      >
        <i
          class="fa-pencil"
        />
        Edit
      </a>
    </li>
    <li
      class="context-menu__item"
    >
      <a
        class="context-menu__link"
        href="##"
      >
        <i
          class="fa-times"
        />
        Delete
      </a>
    </li>
  </ul>
</nav>
`;
